src/Wizard/Services/ConnectivityService.ts

```diff
@@ -0,0 +1,32 @@
+import { Connectivity } from '../Connectivity/Connectivity';
+import { TestType } from '../Models/ConnectivityTest';
+import { inject, injectable, injectAll } from 'tsyringe';
+import { logger } from 'src/Utils';
+import { URL } from 'url';
+
+export const ConnectivityUrls = Symbol('ConnectivityUrls');
+
+@injectable()
+export class ConnectivityService {
+  constructor(
+    @inject(ConnectivityUrls) private readonly options: Map<TestType, URL>,
+    @injectAll(Connectivity)
+    private readonly connectivityTestRegistry: Connectivity[]
+  ) {}
+
+  public async getConnectivityStatus(type: TestType): Promise<boolean> {
```

Comment:
I propose to use better naming for such methods, e.g.
```suggestion
  public async verifyAccess(type: TestType): Promise<boolean> {
```